admin_guide/topics/creating-LC-and-ASG-cluster-auto-scaler.adoc

```diff
@@ -0,0 +1,103 @@
+cml0eS1kYXRhOiAuLi4=
+...
+----
+
+. Create the *_user-data.txt_* cloud-init stanza
++
+[source]
+----
+#cloud-config
+write_files:
+- path: /root/openshift_bootstrap/openshift_settings.yaml
+  owner: 'root:root'
+  permissions: '0640'
+  content: |
+    openshift_node_config_name: node-config-compute 
+- path: /etc/origin/node/bootstrap.kubeconfig
+  owner: 'root:root'
+  permissions: '0640'
+  encoding: b64
+  content: | <1>
+    XBpVmVyc2lvbjogdjEKY2x1c3RlcnM6Ci0gY2x1c3RlcjoKICAgIGNlcnRpZmljYXRlLWF1dGhv
+    cml0eS1kYXRhOiAuLi4=
+
+runcmd:
+- [ ansible-playbook, /root/openshift_bootstrap/bootstrap.yml]
+- [ systemctl, restart, systemd-hostnamed]
+- [ systemctl, restart, NetworkManager]
+- [ systemctl, enable, atomic-openshift-node]
+- [ systemctl, start, atomic-openshift-node]
+----
+<1> Replace this example content with the content from the base64 encoding in the previous step
+
+. Create the Launch Configuration using the AWS CLI
++
+[source,bash]
+----
+aws autoscaling create-launch-configuration \
+    --launch-configuration-name mycluster-LC \ <1>
+    --region us-east-1 \ <2>
+    --image-id ami-987654321 \ <3>
+    --instance-type m4.large \ <4>
+    --security-groups sg-12345678 \ <5>
+    --user-data file://user-data.txt \ <6>
+    --key-name production-key \ <7>
+    --associate-public-ip-address \ <8>
+    --block-device-mappings "[{\"DeviceName\":\"/dev/sda1\",\"Ebs\":{\"SnapshotId\":\"snap-0452f201667751583\"}},{\"DeviceName\":\"/dev/sdb\",\"Ebs\":{\"SnapshotId\":\"snap-09158677d56f88ca7\"}}]" <9>
+----
+<1> The name of the Launch Configuration
+<2> The region to launch the image in
+<3> The primed image AMI
+<4> The type of instance to launch
+<5> The security groups to attach to the launched image
+<6> The cloud-init user-data stanza
+<7> The SSH key-pair name
+<8> Associate a public IP address, optional
+<9> The devices to attach to the image; these can be identified from the AMI image (e.g., ami-987654321) in the AWS console
+
+. Create the Auto Scale Group using the AWS CLI
++
+[source,bash]
+----
+aws autoscaling create-auto-scaling-group \
+      --auto-scaling-group-name mycluster-ASG \ <1>
+      --launch-configuration-name mycluster-LC \ <2>
+      --min-size 0 \ <3>
+      --max-size 6 \ <4>
+      --vpc-zone-identifier subnet-12345678 \ <5>
+      --tags ResourceId=mycluster-ASG,ResourceType=auto-scaling-group,Key=Name,Value=mycluster-ASG-node,PropagateAtLaunch=true ResourceId=mycluster-ASG,ResourceType=auto-scaling-group,Key=kubernetes.io/cluster/mycluster,Value=true,PropagateAtLaunch=true ResourceId=mycluster-ASG,ResourceType=auto-scaling-group,Key=k8s.io/cluster-autoscaler/node-template/label/node-role.kubernetes.io/compute,Value=true,PropagateAtLaunch=true <6>
+----
+<1> the name of the ASG, used later when running the deployment
+<2> the name of the LC, as created in the previous step
+<3> the minimum number of nodes the scale group should maintain
```

Comment:
I need to come back to this; you can increase the size in the web console or through the AWS CLI. (However, I want to investigate what happens if you do this behind the autoscaler's back.) If you increase via the console or CLI you would also have to edit the running deployment:

    oc edit deployment cluster-autoscaler

This will open up an editor and you need to search for the line that reads:

    - --nodes=0:6:mycluster-ASG

and change the min (in this case 0) and the max (in this case 6) to the new values.

This clearly needs a better explanation!